File: hackerrank/algorithms/contests/101hackjun2015/3.py
""" 
    Author: Hemanth Kumar Veeranki
    Handle:harry7

"""
for _ in xrange(input()):
    a,b=raw_input().split()
    if int(a)<int(b):a,b=b,a
    ans=['0']*len(a)
    j=len(a)-1
    for i in xrange(len(b)-1,-1,-1):
        ans[j]=str(min(9,int(b[i])+int(a[j])))
        j-=1
    while j>=0:
        ans[j]=str(int(a[j]))
        j-=1
    if(int("9"*len(a))-int(a)<=int(b)):
        ans="9"*len(a)
        print ans
    else:print ''.join(ans)

File: codeforces/virtual/310/3.java
/* Author: Hemanth Kumar Veeranki */
/* Handle: harry7 */
import java.io.*;
import java.util.*;
import java.math.*;
public class Main{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in); /*sc for taking input*/
		PrintStream op=System.out;/*op for printing output*/
		int n,k,i,j,a=0,b=0,x,y;
		n=sc.nextInt();
		k=sc.nextInt();
		for(i=0;i<k;i++){
			x=sc.nextInt();
			y=sc.nextInt();
			a+=(x-1);
			if(y==1){
				for(j=1;j<x;j++){
					y=sc.nextInt();
					if(y==j+1){
						b=j;
						a--;
					}
				}
			}
			else for(j=1;j<x;j++)y=sc.nextInt();
		}
		op.print(a+n-b-1);
	}
}
